# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: 0552c0b4a079
# Full Hash: 0552c0b4a079a695edfca100df68c4ad82481e58
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Refactor away RtcpEventObserver in favor of MediaEvents. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123223
# ==============================================================================

diff -r 877a0fcc6da9 -r 0552c0b4a079 dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Mon Aug 16 15:57:10 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Thu Aug 19 11:49:29 2021 +0200
@@ -40,8 +40,6 @@
   void OnRtcpBye() override;
   void OnRtcpTimeout() override;
 
-  void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
-
   /**
    * APIs used by the registered external transport to this Conduit to
    * feed in received RTCP Frames to the Receiver for decoding.
@@ -150,6 +148,11 @@
 
   bool IsSamplingFreqSupported(int freq) const override;
 
+  MediaEventSource<void>& RtcpByeEvent() override { return mRtcpByeEvent; }
+  MediaEventSource<void>& RtcpTimeoutEvent() override {
+    return mRtcpTimeoutEvent;
+  }
+
  private:
   WebrtcAudioConduit(const WebrtcAudioConduit& other) = delete;
   void operator=(const WebrtcAudioConduit& other) = delete;
@@ -268,8 +271,9 @@
   // Accessed only on mStsThread
   Maybe<DOMHighResTimeStamp> mLastRtcpReceived;
 
-  // Accessed only on main thread.
-  mozilla::RtcpEventObserver* mRtcpEventObserver = nullptr;
+  // Thread safe
+  MediaEventProducer<void> mRtcpByeEvent;
+  MediaEventProducer<void> mRtcpTimeoutEvent;
 };
 
 }  // namespace mozilla